### ISSUE TITLE
G-Helper preformance is horrible and lag spikes happen when unplugged

### ISSUE THREAD

**Issue by Capn-evil-i** [seerge/g-helper#3291]
### Rules

- [X] I made myself familiar with the <a href='https://github.com/seerge/g-helper?tab=readme-ov-file'>Readme</a>, <a href='https://github.com/seerge/g-helper/wiki/FAQ'>FAQ</a> and <a href='https://github.com/seerge/g-helper/wiki/Troubleshooting'>Troubleshooting</a>.
- [X] I understand that, if insufficient information or no app logs will be provided, my issue will be closed without an answer.

### What's wrong?

I love my ASUS TUF N2NRCX149254082 and i love g helpers' nifty features but when i unplug the laptop compared to armory crate the fps is horrible, it makes my games unplayable because i will go to 120fps and then shoot down to 14-30 every few seconds, on top of this it is the worst on silent mode whilst armory crate has no issues whatsoever, is there a way i can fix this or a config command? Also yes of course i have my GPU on, also i notice its usage is more so 20 percent or a tad bit more in windows task manager

### How to reproduce the bug?

Go to ghelper, unplug it and proceed to set it to silent/ balanced, launch a game, fps is horrible and lags, cant even get consistant 60fps. There is lags spikes and it only reaches optimal fps for less than a second before spiking back down, happens mainly when turning in games but even standing still doing nothing still givres me an issue, i've had this issue in all games, cod, minecraft, iw, portal, scpsl, fortnite, overwatch and pretty much anything else i try

### Logs

[log.txt](https://github.com/user-attachments/files/17495525/log.txt)

### Device and Model

ASUS TUF A15 (I beleive 2022) N2NRCX149254082

### Additional information.

Please give me a response, i am willing to provide any information necessary and i hope you guys can either fix this or help me fix this issue.
Also i checked the requirements and even reinstalled the ASUS system control V3
Also before you come at me, yes i had armory crate uninstalled in the past but i recently reinstalled it because i cannot stand this issue and thought there was something wrong with my laptop originally

### Armoury Crate

Installed

### Asus Services

0/None

### Version

Latest at the time 0.194.0

### OS

Windows 11 24H2 But I've also had issue with 22H2 & 23H2

**Comment by Capn-evil-i** [comment]
The issue is only prominant when ghelper is open by the way to clarify

**Comment by seerge** [comment]
@Capn-evil-i hello,

All 3 modes (Silent / Balanced / Turbo) are baked in firmware / BIOS, so it doesn't matter where you select them - in Armoury or G-Helper. They should behave same unless you explicitly set any custom settings of course. 

It is however normal that device limits power when you play on battery, as this is controlled by firmware.

- What exact mode do you use in Armoury Crate when you play? And how do games behave then ?
- Try to set **same** mode in G-Helper and also click `Factory Defaults` under `Fans+Power` if you have applied some cutomizations.

Thanks

**Comment by Capn-evil-i** [comment]
@seerge Hi, I normally play on performance and I put it on silent during school for obvious reasons, I believe the power is being limited much more in ghelper because I have a slight coil whine and on g helper I hear it come in like bursts kind of and when it comes in for that burst the fps is normal but in armoury crate it’s solid the coil whine in games like it doesn't stop and it also doesent spike fps downward, could you take a look at my logs if you have not already please?

**Comment by Capn-evil-i** [comment]
Also yes I have multiple times tried adjusting the power settings and resetting them to factory default

**Comment by Capn-evil-i** [comment]
I also have tried a reinstall of ghelper

**Comment by seerge** [comment]
@Capn-evil-i 

If you play on `Performance` in Armoury, try to set `Balanced` in G-Helper. It's same thing, just called differently. 

Just don't set any custom fan curves or power limits and click again on `Factory Defaults`. And after that change only one setting under `Fans + Power` – set Power Mode to **Best Performance**. In this situation all settings will be identical to Armoury.

**Comment by Capn-evil-i** [comment]
@seerge thats the issue, I use valences with no extra power settings on a fresh install and it should preform identically and even maybe be quieter but I get the same problem

**Comment by seerge** [comment]
@Capn-evil-i  I have noticed you are using Ambient backlight mode, it's quite resource heavy (as it takes screenshots few times a second in order to adjust keyboard color). Try to set any other backlight mode but not Ambient.

**Comment by Capn-evil-i** [comment]
I am aware that balanced is performance

**Comment by Capn-evil-i** [comment]
@seerge okay I’ll give it a shot in a bit I’ll just set it to color cycle

**Comment by Capn-evil-i** [comment]
@seerge It didn't really work i think it may have helped? but i dont believe so. Does the laptop thermal throttle more unplugged? is there any way i can show you my issue? can i screen record and show you?

**Comment by seerge** [comment]
@Capn-evil-i 
- How do you verify that Performance mode in Armoury Crate works same ? 
- Do you have it installed and running along G-Helper ?

**Comment by Capn-evil-i** [comment]
@seerge I use fps counters in every single game NVIDIA and the game fps counter sometimes, armoury crate doesent drop and I don’t feel it on g helper I can physically feel and see the game freezing up that is why I consider it unplayable. I have it installed but I turn off services when using ghelper

**Comment by seerge** [comment]
@Capn-evil-i G-Helper does not stop Armoury services, it stops only MyASUS related stuff / leftovers (left after AC uninstaller). So in your case Armoury is pretty much functional all the time.

Do a simple test when unplugged without any reboots or so

- Open G-Helper mode to default `Balanced` as I was explaining above and run your game
- Open Armoury and set mode to `Performance and run your game again

Does game behaves differently ?

**Comment by Capn-evil-i** [comment]
@seerge I’m working out, I will test it out and let you know when I get home

**Comment by seerge** [comment]
@Capn-evil-i any updates here ?

**Comment by Capn-evil-i** [comment]
@seerge i am so sorry about that ii've been busy recently, i will try right now

**Comment by Capn-evil-i** [comment]
@seerge No, the issue still persists, as you can see in the video my fps is horrible when i have ghelper open only happens when im on battery, when i use armoury crate the fps unplugged is good again

[****](https://sd40bcca-my.sharepoint.com/:v:/g/personal/1463770_newwestschools_ca/EQ_F7VDwIbJEg3C9DYJ1DgEBkdkybob6GOMpMjp-_88G6g?nav=eyJyZWZlcnJhbEluZm8iOnsicmVmZXJyYWxBcHAiOiJPbmVEcml2ZUZvckJ1c2luZXNzIiwicmVmZXJyYWxBcHBQbGF0Zm9ybSI6IldlYiIsInJlZmVycmFsTW9kZSI6InZpZXciLCJyZWZlcnJhbFZpZXciOiJNeUZpbGVzTGlua0NvcHkifX0&e=OWIP7a)

**Comment by seerge** [comment]
@Capn-evil-i it looks like game just drops FPS when there are other windows on top of it. In your video FPS drops immediately as soon as you open Start menu even (much earlier before launching G-Helper).

You seem to set G-Helper to be always on top of everything under `Extra`. Just go there and uncheck it option and close G-Helper's window (it minimizes in system tray).

And check if it helps.

**Comment by seerge** [comment]
Also did you set `Power mode` under `Fans + Power` to `Best Performance`?

**Comment by Capn-evil-i** [comment]
I get this weird fps on fortnite and games sometimes when i even open ghelper, sorry for the long video but i did everything you said to

https://sd40bcca-my.sharepoint.com/:v:/g/personal/1463770_newwestschools_ca/EUncXM_Z5jdNo-HIgTfJjbQBU4CzTDHIV_hwu6tgvy191w?nav=eyJyZWZlcnJhbEluZm8iOnsicmVmZXJyYWxBcHAiOiJPbmVEcml2ZUZvckJ1c2luZXNzIiwicmVmZXJyYWxBcHBQbGF0Zm9ybSI6IldlYiIsInJlZmVycmFsTW9kZSI6InZpZXciLCJyZWZlcnJhbFZpZXciOiJNeUZpbGVzTGlua0NvcHkifX0&e=bdWwzb

**Comment by Capn-evil-i** [comment]
@seerge Are you sure you wouldn't need any more logs or any of that from me?

**Comment by Capn-evil-i** [comment]
@seerge Also I only get this issue on G-Helper, i am confident in this because i only have this issue when i had it, i reinstalled windows because it drove me crazy and a couple weeks later when i got ghelper i realized that **it is** the problem, before i had it installed there was no issues, i can have it installed, but when it is open then it happens only when unplugged, i do get significantly more fps than armory crate but that's ONLY when i am plugged in, this is slightly annoying because part of a gaming laptop is the portability, do you think this could be a bug in the app on one of these models? or is it too early to say anything?

**Comment by Capn-evil-i** [comment]
@seerge I would not want to use turbo mode because i use my mux switch daily, in the video armoury crate sometimes can't help fix the fps after ghelper takes over even after closing, it is a bit weird

**Comment by Capn-evil-i** [comment]
In the previous video armoury crate hits the normal 120 consistantly
But when i used ghelper it made it 45 ish

**Comment by seerge** [comment]
@Capn-evil-i 

G-helper is not changing settings after closing (it doesn't "run" hardware in real-time, neither does AC). It just selects one of 3 pre-defined modes the moment you click button or moment you start app on the first time.

Turbo is not a mux switch, it's just a bit more performant mode with higher limits. 
But charge your battery a bit before that :) it's already on 23%.

- Can you try to set default Balanced mode in G-Helper then, and after that open Control Panel and select "Performance" power plan there (it should be there cause you also have AC installed, on my screenshot i don't have it cause i don't have AC)
![image](https://github.com/user-attachments/assets/e63a6502-00e3-4df9-883f-e59495491ece)

**Comment by Capn-evil-i** [comment]
@seerge Sure, no problem i'll charge it up a bit and let you know, by the way thank you so much for taking the time to help me with this.

**Comment by seerge** [comment]
@Capn-evil-i and also can you get some CPU specific benchmark (like Cinebench23 or so)

And do one run in Balanced mode in G-Helper and after that - same run in Performance mode in Armoury. And check the scores, are they different ?

**Comment by Capn-evil-i** [comment]
> @Capn-evil-i and also can you get some CPU specific benchmark (like Cinebench23 or so)
> 
> And do one run in Balanced mode in G-Helper and after that - same run in Performance mode in Armoury. And check the scores, are they different ?

Yeah i also believe that it is a cpu issue, sure i'll download cinebench in the meantime

**Comment by seerge** [comment]
@Capn-evil-i did you had a chance to run CPU benchmark to compare scores ?

**Comment by Capn-evil-i** [comment]
@seerge I'm so sorry I've been busy with school work and work, ill find a time on the weekend to benchmark it for you

**Comment by seerge** [comment]
@Capn-evil-i any updates here ?

**Comment by Capn-evil-i** [comment]
> @Capn-evil-i any updates here ?

He seerge, after some experimenting, ii realized that my fps is normally this low, only for a few seconds after i switch the mode in AC it's this fps, thank you for helping though, sorry for the long wait.

**Comment by seerge** [comment]
@Capn-evil-i I didn’t completely get you. So when you set mode in AC FPS also drops?

**Comment by Capn-evil-i** [comment]
> @Capn-evil-i I didn’t completely get you. So when you set mode in AC FPS also drops?

Nevermind sorry the fps is a lot worse on helper unplugged. I’ll actually do some benchmarks today for you

**Comment by seerge** [comment]
@Capn-evil-i hello, do you have any updates ? Did you manage to run cinebench23 in AC and G-Helper ?

I have something for you to try. Download latest release https://github.com/seerge/g-helper/releases/tag/v0.200 open `Fans+Power` and select newly added option in `Power mode` dropdown - `High Performance Plan`. And check if it makes difference on your games.

Thanks

**Comment by seerge** [comment]
Closed as abandoned
